### PR TITLE
Feature/button

### LINKED ISSUE #2684: Refactor Button component to App Components

### PR DESCRIPTION
**Pull request by HauklandJ**
## Description
Creates button wrapper for Designsystemet button at replaces usage in the rest of the app.

## Related Issue(s)

- closes #2684 

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [x] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [x] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [ ] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Review by Magnusrm — APPROVED** [review]
Great work! 👏

File: src/app-components/button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      size={size}
+      ref={ref}
+      {...props}
+    >
+      {isLoading && (
+        <Spinner
+          aria-hidden='true'
+          color={color}
+          size={size === 'lg' ? 'sm' : 'xs'}
+          title='Laster innhold'
```
Comment (olemartinorg) [inline src/app-components/button/Button.tsx:56]: This should be localized - our apps support multiple languages. :raised_hands:

File: src/layout/ActionButton/ActionButtonComponent.tsx [hunk 2]
```diff
@@ -1,18 +1,14 @@
 import React from 'react';
 
-import { Button } from '@digdir/designsystemet-react';
-
 import type { PropsFromGenericComponent } from '..';
 
+import { Button, type ButtonColor, type ButtonVariant } from 'src/app-components/button/Button';
 import { useProcessNavigation } from 'src/features/instance/ProcessNavigationContext';
 import { Lang } from 'src/features/language/Lang';
-import { ButtonLoader } from 'src/layout/Button/ButtonLoader';
```
Comment (olemartinorg) [inline src/layout/ActionButton/ActionButtonComponent.tsx:9]: I see the `ButtonLoader` component is no longer in use now, so it should probably be deleted. :wastebasket:

File: src/layout/CustomButton/CustomButtonComponent.tsx [hunk 3]
```diff
@@ -214,8 +214,25 @@ export const buttonStyles: { [style in CBTypes.ButtonStyle]: { color: ButtonColo
   secondary: { variant: 'secondary', color: 'first' },
 };
 
+function toShorthandSize(size?: CBTypes.ButtonSize): 'sm' | 'md' | 'lg' | undefined {
```
Comment (olemartinorg) [inline src/layout/CustomButton/CustomButtonComponent.tsx:217]: Nitpick: Seems to me like this cannot really return `undefined`. Simply removing the return type definition here makes typescript infer the correct return type of `'sm' | 'md' | 'lg'`.

File: test/e2e/integration/frontend-test/summary.ts [hunk 4]
```diff
@@ -65,7 +65,8 @@ describe('Summary', () => {
     cy.dsSelect(appFrontend.changeOfName.uploadWithTag.tagsDropDown, 'Adresse');
     cy.get(appFrontend.changeOfName.uploadWithTag.saveTag).click();
 
-    cy.get(appFrontend.backToSummaryButton).click();
+    // cy.findByRole( 'button', { name: /Tilbake til oppsummering/ }).click();
```
Comment (olemartinorg) [inline test/e2e/integration/frontend-test/summary.ts:68]: Nitpick: Could probably just be deleted instead of commented-out?

**Review by olemartinorg — APPROVED** [review]
Nice work! :star_struck: Selecting by roles in all the tests must have been quite a chore, but now we're much better off. :pray:

**Comment by olemartinorg** [comment]
Sorry! The failing cypress test here is my fault (`group-pets.ts`). It will be fixed in #2690.

**Pull request merged**